devito/finite_difference.py

```diff
@@ -31,6 +32,24 @@
 h = symbols('h')
 
 
+class Transpose(object):
+    """Class that defines if the derivative is itself or adjoint (transpose).
+    This only matter for odd order derivatives that requires
+    a minus sign for the transpose."""
+    def __init__(self, transpose):
+        self._transpose = transpose
+
+    def __eq__(self, other):
+        return self._transpose == other._transpose
+
+    def __repr__(self):
+        return {1: 'direct', -1: 'transpose'}[self._transpose]
+
+
+direct = Transpose(1)
+transpose = Transpose(-1)
+
```

Comment:
Perhaps expressible in a more concise way as:
`from collections import namedtuple

Transpose = namedtuple('Transpose', 'transpose')
DIRECT = Transpose(1)
TRANSPOSE = Transpose(-1)
`
?